moxy-compiler/src/main/java/moxy/compiler/viewstateprovider/ViewStateProviderClassGenerator.kt

```diff
@@ -15,9 +15,13 @@ import moxy.compiler.toJavaFile
 import moxy.viewstate.MvpViewState
 import javax.lang.model.element.Modifier
 
-class ViewStateProviderClassGenerator : JavaFilesGenerator<PresenterInfo> {
+class ViewStateProviderClassGenerator : JavaFilesGenerator<PresenterInfo?> {
+
+    override fun generate(presenterInfo: PresenterInfo?): List<JavaFile> {
+        if (presenterInfo == null) {
+            return emptyList()
+        }
```

Comment:
I don't think this class should handle nulls. Also, as can be seen in `MvpCompiler.generateCode()`, we do not call `ViewStateProviderClassGenerator` if `InjectViewStateProcessor` returned null.

Reply:
We should not, but the fact that `InjectViewStateProcessor` now returns `PresenterInfo?` forces us to use `PresenterInfo?` as a generic parameter in `ViewStateProviderClassGenerator`. If we write `class ViewStateProviderClassGenerator : JavaFilesGenerator<PresenterInfo>`, `processInjectors` invocation breaks at `MvpCompiler:104`. This is subject for refactoring, but not for now, I think.